refactor AuroraDB tests and add migration-tests to CI job

kami619 opened the pull request:
Fixes #31024 

**Background**:
Earlier this year, [from this change](https://github.com/keycloak/keycloak/pull/26970), we added a new matrix option to run the `store-integration-tests`. And now as per #31024 we need to run tests against the AuroraDB again. Given this and from the discussions with @ryanemerson and @mhajas we understood, it would be best to refactor the `store-integration-tests` and create a new job for the tests to be run against AuroraDB, which would allow us to optimize around the cost and code.

This PR refactors the Keycloak CI workflow for the bits associated with the AuroraDB database. We are adding a new job named `auroradb-integration-tests` to the `jobs`. 

Test run on my fork: https://github.com/kami619/keycloak/actions/runs/9979098679/job/27577694974

Note: When there is a change to `store-integration-tests` or `migration-tests`, we would need to update the `auroradb-integration-tests` job as well.

kami619 committed:
refactor auroradb tests from store-integration-tests

Signed-off-by: Kamesh Akella <kamesh.asp@gmail.com>

ryanemerson commented on .github/workflows/ci.yml:
```diff
@@ -485,19 +475,76 @@ jobs:
+          export CLUSTER_NAME=${{ steps.aurora-migration-tests.outputs.ec2_cluster }}
+          ./aws_ec2.sh delete ${{ steps.aurora-init.outputs.region }}
+
+      - name: Run Aurora integration tests on EC2
```
There's a lot of duplication here, is there any reason why we can't re-use the same ec2 instance for executing both the migration and store intgration tests?

I think it should be possible to share all of the preliminary steps between the two test types, e.g. do the following in a single step:

1. Download aws.pem
2. Provision ec2
3. install and build keycloak

We can then have dedicated steps for executing the two different `./mvn_ec2_runner.sh ${REGION} "test ...` commands and retrieving the output.

kami619 replied:
I was thinking about that too @ryanemerson. Let me give it a try. Thanks for the suggestion.

kami619 replied:
@ryanemerson I think the latest should be as per what we discussed. lmk WDYT ?

ryanemerson commented on .github/workflows/ci.yml:
```diff
@@ -419,68 +421,51 @@ jobs:
-          echo "Tests: $TESTS"
-
+          PROPS="-Dkeycloak.connectionsJpa.jdbcParameters=\"?ssl=true&sslmode=verify-ca&sslrootcert=/opt/keycloak/aws.pem\""
+          echo "maven_properties=${PROPS}" >> $GITHUB_OUTPUT
```
As the properties are related to aurora, not ec2, I think it makes sense for the maven_properties output to be set in the `aurora-init` step. For the same reason, we can also move the `curl` command to the `aurora-init` step.

kami619 replied:
this should be fixed now.

ryanemerson reviewed (commented):
The changes are looking good to me, just a couple of minor comments about code cleanup.

kami619 commented on .github/workflows/ci.yml:
```diff
@@ -489,19 +474,71 @@ jobs:
+          REGION=${{ steps.aurora-init.outputs.region }}
+          PROPS=${{ steps.ec2-create.outputs.maven_properties }}
+          PROPS+=" -Dauth.server.db.host=${{ steps.aurora-create.outputs.endpoint }}"
+          PROPS+=" -Dkeycloak.connectionsJpa.password=${{ steps.aurora-init.outputs.password }}"
```
I can only move to set the host value in `maven_properties` to the `aurora-init` step, as the password masking is preventing the translation of the value properly and resulting in the below error.

```
/home/runner/work/_temp/b8943753-ce47-4964-8ed8-3bf249440e9a.sh: line 3: -Dkeycloak.connectionsJpa.*** command not found
Error: Process completed with exit code 127.
```

kami619 commented:
@ryanemerson the code should be ready for review again, however, I will update this comment to include a successful run against my fork. Thanks for your patience in reviewing this.

kami619 commented:
@ryanemerson https://github.com/kami619/keycloak/actions/runs/10013614481/job/27681805510

For some reason, if I split the handling of aws.pem file across steps, it fails with the below exception.

https://github.com/kami619/keycloak/actions/runs/10013614481/job/27681805510

And it doesn't happen when I changed it back to handling aws.pem within the same step, am I missing something here ?

mhajas commented:
Moving to draft as this is still in progress

kami619 commented:
https://github.com/kami619/keycloak/actions/runs/10065470972/job/27824659034

@mhajas and @ryanemerson I think this should be in a good state to review now.

ryanemerson commented on .github/workflows/ci.yml:
```diff
@@ -420,89 +422,126 @@ jobs:
           git archive --format=zip --output /tmp/keycloak.zip $GITHUB_REF
           zip -u /tmp/keycloak.zip aws.pem
-
+          zip -ur /tmp/keycloak.zip .git
```
We should be able to remove this line. I added this commit as I thought it would fix a warning with the buildnumber-maven-plugin, but that warning turns out not to be related to the issues we were having and this change didn't actually fix the issue :sweat_smile:

kami619 replied:
sure thing Ryan. I will push that change.

ryanemerson reviewed (approved):
LGTM, thanks for persevering @kami619


There's one minor comment, but consider it optional.

mhajas reviewed (approved):
Thank you @kami619! LGTM
